drivers/atwinc15x0/Makefile.dep

```diff
@@ -1,6 +1,14 @@
 USEMODULE += netdev_eth
 USEMODULE += ztimer_msec
 USEMODULE += netdev_legacy_api
+
+ifneq (,$(filter atwinc15x0_dynamic_scan,$(USEMODULE)))
```

Comment:
When I tried to test this PR the first time, I realized that `examples/gnrc_networking` doesn't work out of the box anymore. To get it working, module `atwinc15x0_static_connect` has to be enabled explicitly. Therefore, I suggest to enable `atwinc15x0_static_connect` by default if no other module is enabled:
```suggestion
ifeq (,$(filter atwinc15x0_dynamic_%,$(USEMODULE)))
  USEMODULE += atwinc15x0_static_connect
endif

ifneq (,$(filter atwinc15x0_dynamic_scan,$(USEMODULE)))
```

Reply:
What about this change to enable `atwinc15x0_static_connect` if none of the `atwinc15x0_dynamic_*` modules is enabled?

Reply:
> It´s [574191e](https://github.com/RIOT-OS/RIOT/commit/574191e583343d222d84b8ce2961af5fcdbfb1f7)
> 
> Are fixups getting messy? Do you want me to squash?

Ok

Reply:
> It´s [574191e](https://github.com/RIOT-OS/RIOT/commit/574191e583343d222d84b8ce2961af5fcdbfb1f7)

Sorry again for asking again for this change. Now I know why I thought that this issue wasn't fix. Usually, GitHub labels files for which a new version is uploaded as `Outdated`. However, this file isn't labeled as outdated and if you click on the file name, it still openes the initial version. That's why I didn't realized that you changed it already.